info.limpet.test/src/info/limpet/data/TestOperations.java

```diff
@@ -800,6 +803,39 @@ public void testSubtraction()
   }
 
   @SuppressWarnings("unchecked")
+  public void testAddLayerOperation()
+  {
+    // place to store results data
+    InMemoryStore store = new SampleData().getData(10);
+
+    List<IStoreItem> selection = new ArrayList<IStoreItem>();
+
+    StoreGroup track1 = new StoreGroup("Track 1");
+    selection.add(track1);
+
+    Collection<ICommand<IStoreItem>> commands =
+        new AddLayerOperation().actionsFor(selection, store, context);
+    assertEquals("invalid number of inputs", 1, commands.size());
+    for (ICommand<IStoreItem> iCommand : commands)
+    {
+      iCommand.execute();
+    }
+
+    IQuantityCollection<Velocity> speedGood1 =
+        (IQuantityCollection<Velocity>) store.get(SampleData.SPEED_ONE);
+    selection = new ArrayList<>();
+    selection.add(speedGood1);
+
+    commands = new AddLayerOperation().actionsFor(selection, store, context);
+    assertEquals("invalid number of inputs", 1, commands.size());
+    for (ICommand<IStoreItem> iCommand : commands)
+    {
+      iCommand.execute();
+      iCommand.dataChanged(speedGood1);
+    }
+  }
```

Comment:
from looking at the AddLayerOperation class, I can see that if there isn't anything provided in the selection, then a new layer is created at the top level of the store.